lib/services/role.go

```diff
@@ -627,6 +636,32 @@ func (r *RoleV3) SetNodeLabels(rct RoleConditionType, labels Labels) {
 	}
 }
 
+// GetClusterLabels gets the map of cluster labels this role is allowed or denied access to.
+func (r *RoleV3) GetClusterLabels(rct RoleConditionType) Labels {
+	if rct == Allow {
+		return defaultAllowClusterLabels(r.Spec.Allow.ClusterLabels)
```

Comment:
IIUC, this will result in a user having wildcard `allow` so long as *any* role does not specify allow labels.  This is problematic, as adding a role will effectively be able to silently add allow directives.

What about either defaulting to a wildcard at the `RoleSet` level (i.e. assume wilcard until at least one of a user's roles includes an `allow` directive), or simply assume that all users have access to unlabeled clusters?  Both strategies should be backwards-compatible, but don't silently grant new privileges when roles are added to existing users.